allensdk/brain_observatory/behavior/behavior_project_cache/project_apis/data_io/natural_movie_one_cache.py

```diff
@@ -0,0 +1,95 @@
+from typing import Union
+from allensdk.api.cloud_cache.cloud_cache import S3CloudCache
+import pathlib
+from allensdk.api.cloud_cache.file_attributes import CacheFileAttributes
+from allensdk.brain_observatory.behavior.data_objects.stimuli.stimulus_templates import StimulusMovieTemplateFactory  # noqa: E501
+import numpy as np
+
+
+class NaturalMovieOneCache(S3CloudCache):
+
+    def __init__(self,
+                 cache_dir: Union[str, pathlib.Path],
+                 bucket_name: str):
+        super().__init__(cache_dir=cache_dir,
+                         bucket_name=bucket_name,
+                         project_name=None,
+                         ui_class_name=None)
+
+        # Set the file attributes by hand. This is used to get around needing
+        # to run the data release tool and create/download a manifest file.
+        # The hash has been pre-calculated from the file_hash_from_path
+        # method in allensdk/api/cloud_cache/utils.py
+        self._file_attributes = CacheFileAttributes(
+            url="https://staging.visual-behavior-ophys-data.s3.us-west-2.amazonaws.com/visual-behavior-ophys/resources/Movie_TOE1.npy", # noqa E501
+            version_id="0y.DEg5ASDGaWA4Syls5MeC.S5Y6oIIS",
+            file_hash='7e44cba154b29e1511ab8e5453b7aa5070f1ae456724b5b2541c97c052fbd80aebf159e5f933ab319bda8fdab7b863a096cdb44f129abd20a8c4cc791af4bc41', # noqa E501
+            local_path=pathlib.Path(cache_dir) / "Movie_TOE1.npy"
+        )
+
+    def _list_all_manifests(self) -> list:
+        """
+        Return a list of all of the file names of the manifests associated
+        with this dataset
+        """
+        return None
+
+    def download_data(self) -> pathlib.Path:
```

Comment:
Is this functionality different than the `download_data` in the base class?

Reply:
Slightly, the ordering is different in that things not used because the file metadata is set and doesn't need to be retrieved. So it's basically as simpler version of what is there.

Reply:
It seems that you instead want to override `data_path` and then the `download_data` of the superclass will just work? Want to try to get rid of the bit of boilerplate code duplication here.